### ISSUE TITLE
Shiny: biclust example app not rendering locally

### ISSUE THREAD

**Issue by bborgesr** [rstudio/flexdashboard#158]
The [3rd example](https://jjallaire.shinyapps.io/shiny-biclust/) in the flexdashboard website [Examples tab](http://rmarkdown.rstudio.com/flexdashboard/examples.html) is no longer rendering when I try to run it locally. It simply opens up a window with the dreaded red message: "Error: pandoc document conversion failed with error 1" 

I don't know how long ago this started, but it happens in Pandoc 1.9 as well, so it's not related to the release of Pandoc 2.0.

Since this has been broken a while without us noticing, I'll assume it's low priority and move on for now.

<details>
  <summary>app.Rmd</summary>

````julia
---
title: "Gene Expression Biclustering"
author: "Bryan Lewis"
output: 
  flexdashboard::flex_dashboard:
    orientation: rows
    social: menu
    source_code: embed
runtime: shiny
---

```{r global, include=FALSE}
# load data in 'global' chunk so it can be shared by all users of the dashboard
library(biclust)
data(BicatYeast)
set.seed(1)
res <- biclust(BicatYeast, method=BCPlaid(), verbose=FALSE)
```

Inputs {.sidebar}
-----------------------------------------------------------------------

```{r}
 selectInput("clusterNum", label = h3("Cluster number"), 
    choices = list("1" = 1, "2" = 2, "3" = 3, "4" = 4, "5" = 5), 
    selected = 1)
```

Microarray data matrix for 80 experiments with Saccharomyces Cerevisiae
organism extracted from R's `biclust` package.

Sebastian Kaiser, Rodrigo Santamaria, Tatsiana Khamiakova, Martin Sill, Roberto
  Theron, Luis Quintales, Friedrich Leisch and Ewoud De Troyer. (2015). biclust:
  BiCluster Algorithms. R package version 1.2.0.
  http://CRAN.R-project.org/package=biclust

Row
-----------------------------------------------------------------------

### Heatmap
    
```{r}

num <- reactive(as.integer(input$clusterNum))

col = colorRampPalette(c("red", "white", "darkblue"), space="Lab")(10)
renderPlot({
    p = par(mai=c(0,0,0,0))
    heatmapBC(BicatYeast, res, number=num(), xlab="", ylab="",
      order=TRUE, useRaster=TRUE, col=col)
    par(p)
})
```


Row {.tabset}
-----------------------------------------------------------------------

### Parallel Coordinates
    
```{r}
renderPlot(
  parallelCoordinates(BicatYeast, res, number=num())
)
```
   
### Data for Selected Cluster

```{r}
# only display table for values in cluster 4
renderTable(
  BicatYeast[which(res@RowxNumber[, num()]), which(res@NumberxCol[num(), ])]
)
```
````

</details>

<details>
  <summary>Rmarkdown console output</summary>

```r
processing file: shiny-biclust-example.Rmd
  |......                                                           |  10%
  ordinary text without R code

  |.............                                                    |  20%
label: global (with options) 
List of 1
 $ include: logi FALSE

Loading required package: MASS
Loading required package: grid
Loading required package: colorspace
Loading required package: lattice
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
  |....................                                             |  30%
  ordinary text without R code

  |..........................                                       |  40%
Warning in seq.default(along = x) :
  partial argument match of 'along' to 'along.with'
label: unnamed-chunk-1
  |................................                                 |  50%
  ordinary text without R code

  |.......................................                          |  60%
label: unnamed-chunk-2
  |..............................................                   |  70%
  ordinary text without R code

  |....................................................             |  80%
label: unnamed-chunk-3
  |..........................................................       |  90%
  ordinary text without R code

  |.................................................................| 100%
label: unnamed-chunk-4

output file: /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/Rtmpjg31k0/shiny-biclust-example.knit.md

/usr/local/bin/pandoc +RTS -K512m -RTS /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/Rtmpjg31k0/shiny-biclust-example.utf8.md --to html4 --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output /private/var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T/Rtmpjg31k0/file4dac6583fe91.html --email-obfuscation none --standalone --section-divs --template /Library/Frameworks/R.framework/Versions/3.4/Resources/library/flexdashboard/rmarkdown/templates/flex_dashboard/resources/default.html --include-in-header /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/rmarkdown-str4dac3cbe15a8.html --id-prefix section- --variable 'theme:cosmo' --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/file4dac1b5d7345html --include-after-body /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/file4dac56a96c7f.html --highlight-style pygments --include-before-body /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/file4dac143ac50c.html --include-after-body /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/file4dac1e5ea134.html 
pandoc: /var/folders/6b/xmhjngk91hd6d61027q69kz40000gn/T//Rtmpjg31k0/rmarkdown-str4dac3cbe15a8.html: openBinaryFile: does not exist (No such file or directory)
Warning: Error in : pandoc document conversion failed with error 1
Stack trace (innermost first):
    100: pandoc_convert
     99: convert
     98: <Anonymous>
     97: do.call
     96: contextFunc
     95: .getReactiveEnvironment()$runWith
     94: shiny::maskReactiveContext
     93: <reactive>
     82: doc
     81: shiny::renderUI
     80: func
     79: origRenderFunc
     78: output$__reactivedoc__
      3: <Anonymous>
      2: do.call
      1: rmarkdown::run
```

</details>

<details>
  <summary><code>devtools::session_info()</code></summary>

```r
> devtools::session_info()
Session info -----------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, darwin15.6.0        
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       Europe/Lisbon               
 date     2017-11-17                  

Packages ---------------------------------------------------------------------------------------------
 package       * version date       source                      
 backports       1.1.1   2017-09-25 CRAN (R 3.4.2)              
 base          * 3.4.2   2017-10-04 local                       
 compiler        3.4.2   2017-10-04 local                       
 datasets      * 3.4.2   2017-10-04 local                       
 devtools        1.13.4  2017-11-09 CRAN (R 3.4.2)              
 digest          0.6.12  2017-01-27 CRAN (R 3.4.0)              
 evaluate        0.10.1  2017-06-24 CRAN (R 3.4.1)              
 flexdashboard   0.5     2017-11-17 local                       
 graphics      * 3.4.2   2017-10-04 local                       
 grDevices     * 3.4.2   2017-10-04 local                       
 htmltools       0.3.6   2017-04-28 CRAN (R 3.4.0)              
 jsonlite        1.5     2017-06-01 CRAN (R 3.4.0)              
 knitr           1.17.12 2017-11-17 Github (yihui/knitr@b2d8929)
 magrittr        1.5     2014-11-22 CRAN (R 3.4.0)              
 memoise         1.1.0   2017-04-21 CRAN (R 3.4.0)              
 methods       * 3.4.2   2017-10-04 local                       
 Rcpp            0.12.13 2017-09-28 cran (@0.12.13)             
 rmarkdown       1.7     2017-11-10 CRAN (R 3.4.2)              
 rprojroot       1.2     2017-01-16 CRAN (R 3.4.0)              
 rsconnect       0.8.5   2017-08-23 CRAN (R 3.4.1)              
 stats         * 3.4.2   2017-10-04 local                       
 stringi         1.1.5   2017-04-07 cran (@1.1.5)               
 stringr         1.2.0   2017-02-18 CRAN (R 3.4.0)              
 tools           3.4.2   2017-10-04 local                       
 utils         * 3.4.2   2017-10-04 local                       
 withr           2.1.0   2017-11-01 CRAN (R 3.4.2)              
 yaml            2.1.14  2016-11-12 CRAN (R 3.4.0)              
```
</details>

**Comment by bborgesr** [comment]
The exact same thing happens with the 4th app (the CRAN download monitor). Here's the code for convenience:

<details>
  <summary>shiny-cran-downloads-functions.R</summary>

```r
library(shiny)
library(shinySignals)
library(dplyr)
library(bubbles)

# An empty prototype of the data frame we want to create
prototype <- data.frame(date = character(), time = character(),
  size = numeric(), r_version = character(), r_arch = character(),
  r_os = character(), package = character(), version = character(),
  country = character(), ip_id = character(), received = numeric())

# Connects to streaming log data for cran.rstudio.com and
# returns a reactive expression that serves up the cumulative
# results as a data frame
packageStream <- function(session = getDefaultReactiveDomain()) {
  # Connect to data source
  sock <- socketConnection("cransim.rstudio.com", 6789, blocking = FALSE, open = "r")
  # Clean up when session is over
  session$onSessionEnded(function() {
    close(sock)
  })
  
  # Returns new lines
  newLines <- reactive({
    invalidateLater(1000, session)
    readLines(sock)
  })
  
  # Parses newLines() into data frame
  reactive({
    if (length(newLines()) == 0)
      return()
    read.csv(textConnection(newLines()), header=FALSE, stringsAsFactors=FALSE,
      col.names = names(prototype)
    ) %>% mutate(received = as.numeric(Sys.time()))
  })
}

# Accumulates pkgStream rows over time; throws out any older than timeWindow
# (assuming the presence of a "received" field)
packageData <- function(pkgStream, timeWindow) {
  shinySignals::reducePast(pkgStream, function(memo, value) {
    rbind(memo, value) %>%
      filter(received > as.numeric(Sys.time()) - timeWindow)
  }, prototype)
}

# Count the total nrows of pkgStream
downloadCount <- function(pkgStream) {
  shinySignals::reducePast(pkgStream, function(memo, df) {
    if (is.null(df))
      return(memo)
    memo + nrow(df)
  }, 0)
}

# Use a bloom filter to probabilistically track the number of unique
# users we have seen; using bloom filter means we will not have a
# perfectly accurate count, but the memory usage will be bounded.
userCount <- function(pkgStream) {
  # These parameters estimate that with 5000 unique users added to
  # the filter, we'll have a 1% chance of false positive on the next
  # user to be queried.
  bloomFilter <- BloomFilter$new(5000, 0.01)
  total <- 0
  reactive({
    df <- pkgStream()
    if (!is.null(df) && nrow(df) > 0) {
      # ip_id is only unique on a per-day basis. To make them unique
      # across days, include the date. And call unique() to make sure
      # we don't double-count dupes in the current data frame.
      ids <- paste(df$date, df$ip_id) %>% unique()
      # Get indices of IDs we haven't seen before
      newIds <- !sapply(ids, bloomFilter$has)
      # Add the count of new IDs
      total <<- total + length(newIds)
      # Add the new IDs so we know for next time
      sapply(ids[newIds], bloomFilter$set)
    }
    total
  })
}

# Quick and dirty bloom filter. The hashing "functions" are based on choosing
# random sets of bytes out of a single MD5 hash. Seems to work well for normal
# values, but has not been extensively tested for weird situations like very
# small n or very large p.

library(digest)
library(bit)

BloomFilter <- setRefClass("BloomFilter",
  fields = list(
    .m = "integer",
    .bits = "ANY",
    .k = "integer",
    .bytesNeeded = "integer",
    .bytesToTake = "matrix"
  ),
  methods = list(
    # @param n - Set size
    # @param p - Desired false positive probability (e.g. 0.01 for 1%)
    initialize = function(n = 10000, p = 0.001) {
      m = (as.numeric(n) * log(1 / p)) / (log(2)^2)
      
      .m <<- as.integer(m)
      .bits <<- bit(.m)
      .k <<- max(1L, as.integer(round((as.numeric(.m)/n) * log(2))))
      
      # This is how many *bytes* of data we need for *each* of the k indices we need to
      # generate
      .bytesNeeded <<- as.integer(ceiling(log2(.m) / 8))
      .bytesToTake <<- sapply(rep_len(.bytesNeeded, .k), function(byteCount) {
        # 16 is number of bytes an md5 hash has
        sample.int(16, byteCount, replace = FALSE)
      })
    },
    .hash = function(x) {
      hash <- digest(x, "md5", serialize = FALSE, raw = TRUE)
      sapply(1:.k, function(i) {
        val <- rawToInt(hash[.bytesToTake[,i]])
        # Scale down to fit into the desired range
        as.integer(val * (as.numeric(.m) / 2^(.bytesNeeded*8)))
      })
    },
    has = function(x) {
      all(.bits[.hash(x)])
    },
    set = function(x) {
      .bits[.hash(x)] <<- TRUE
    }
  )
)

rawToInt <- function(bytes) {
  Reduce(function(left, right) {
    bitwShiftL(left, 8) + right
  }, as.integer(bytes), 0L)
}
```

</details> 

<details>
  <summary>app.Rmd</summary>
  
````julia
---
title: "CRAN Download Monitor"
output: 
  flexdashboard::flex_dashboard:
    orientation: rows
    social: menu
    source_code: https://git.io/vaZdx
runtime: shiny
---

```{r setup, include=FALSE}
library(flexdashboard)

# Access to CRAN packages data stream
source("shiny-cran-downloads-functions.R")

# pkgStream is a reactive expression that represents a stream of
# new package download data; up to once a second it may return a
# data frame of new downloads since the last update.
pkgStream <- packageStream()

# pkgData is a reactive expression that accumulates previous values
# of pkgStream, discarding any that are older than maxAgeSecs.
maxAgeSecs <- 60 * 5 
pkgData <- packageData(pkgStream, maxAgeSecs)
```

Sidebar {.sidebar}
=======================================================================

### cran.rstudio.com

The streaming data is a 1-week-delayed livestream of download
logs from  cran.rstudio.com. The server code for that service is at [jcheng5/cransim](https://github.com/jcheng5/cransim).


```{r}
# Rate at which to flag high download traffic
sliderInput("rateThreshold", "Warn when rate exceeds:",
            min = 0, max = 50, value = 3, step = 0.1
)

# Maximum number of raw data rows to keep
numericInput("maxrows", "Recent downloads window:", 50)
```


Dashboard
=======================================================================

Row
-----------------------------------------------------------------------

### Downloads per sec (last 5 min) {.value-box}

```{r}
# downloadRate is a reactive expression that computes the download
# rate during this dashboard's lifetime.
startTime <- as.numeric(Sys.time())
downloadRate <- reactive({
  elapsed <- as.numeric(Sys.time()) - startTime
  nrow(pkgData()) / min(maxAgeSecs, elapsed)
})

# Emit the download rate
renderValueBox({
  rate <- formatC(downloadRate(), digits = 1, format = "f")
  valueBox(
    value = rate,
    icon = "fa-area-chart",
    color = if (rate >= input$rateThreshold) "warning" else "primary"
  )
})
```

### Total downloads {.value-box}

```{r}
# dlCount is a reactive expression that keeps track of the total
# number of rows that have ever appeared through pkgStream.
dlCount <- downloadCount(pkgStream)

# Emit the download count
renderValueBox({
  valueBox(dlCount(), icon = "fa-download")
})
```

### Unique users {.value-box}

```{r}
# usrCount is a reactive expression that keeps an approximate
# count of all of the unique users that have been seen since the
# app started.
usrCount <- userCount(pkgStream)

# Emit the user count
renderValueBox({
  valueBox(value = usrCount(), icon = "fa-users")
})
```

Row
-----------------------------------------------------------------------

### Popularity by package (last 5 min) {data-width=700}

```{r}
renderBubbles({
  if (nrow(pkgData()) == 0)
    return()

  order <- unique(pkgData()$package)
  df <- pkgData() %>%
    group_by(package) %>%
    tally() %>%
    arrange(desc(n), tolower(package)) %>%
    # Just show the top 60, otherwise it gets hard to see
    head(60)

  bubbles(df$n, df$package, key = df$package)
})
```

### Percent of downloads (last 5 min) {data-width=340}

```{r}
renderTable({
  pkgData() %>%
    group_by(package) %>%
    tally() %>%
    arrange(desc(n), tolower(package)) %>%
    mutate(percentage = n / nrow(pkgData()) * 100) %>%
    select("Package" = package, "Percent" = percentage) %>%
    as.data.frame() %>%
    head(30)
}, digits = 1)

```

Recent Downloads
=======================================================================

### Recent Downlads

```{r}
renderTable({
  downloads <- tail(pkgData(), n = input$maxrows)
  downloads <- downloads[,c("date", "time", "size", "r_version", 
                            "r_arch", "r_os", "package")]
  downloads[order(nrow(downloads):1),]
})
```

````

</details>

**Comment by bborgesr** [comment]
When trying to find a minimal repro, I found that removing the part:

````yaml
    social: menu
    source_code: embed
````
from the YAML header, the document does render. But it doesn't know what `input` and `output` are. These seem two distinct bugs and the latter is the scarier one. I've noticed that all the dashboards that have been failing in this way are `runtime: shiny` ones. It seems that ANY `runtime: shiny` document is vulnerable and that this bug manifests whenever the author interacts with `input` or `output`. See https://github.com/rstudio/rmarkdown/issues/1206.
